src/server/game/Handlers/MovementHandler.cpp

```diff
@@ -608,3 +608,27 @@ void WorldSession::HandleSummonResponseOpcode(WorldPacket& recvData)
 
     _player->SummonIfPossible(agree);
 }
+
+void WorldSession::HandleMoveTimeSkippedOpcode(WorldPacket& recvData)
+{
+    TC_LOG_DEBUG("network", "WORLD: Received CMSG_MOVE_TIME_SKIPPED");
+
+    ObjectGuid guid;
+    uint32 timeSkipped;
+    recvData >> guid.ReadAsPacked();
+    recvData >> timeSkipped;
+
+    Unit* mover = GetPlayer()->m_unitMovedByMe;
+    ASSERT(mover != nullptr);                      // there must always be a mover
```

Comment:
can't this just be a null check ? why does it have to be an assert ?

Reply:
You must NEVER assert in client packet handlers

Reply:
Indeed. Good point. In that case though, existing code will need to be modified as well since there already are a few asserts in MovementHandler.

Reply:
half of "crashes" we get seems to be triggered asserts :P

Reply:
Well that's a good thing, they do outline abnormal behavior instead of failing silently

Reply:
@kelno well yes and no. the issue is that an attacker could, by sending custom packets, intentionally crash a server running on TC. Example here: If a client sends a CMSG_MOVE_TIME_SKIPPED packet while his character is feared, the condition `mover != nullptr` won't be met and the server will crash.

Reply:
Yes in this case it's totally bad I agree, I was making a more general comment in response for the last comment!